ement-room.el

```diff
@@ -2682,12 +2688,19 @@ Format defaults to `ement-room-message-format-spec', which see."
              (propertize " "
                          'display `((margin right-margin) ,string))))))
       (when-let* ((data (ement-event-local event))
-                  (num (map-elt data 'url-preview-n)))
+                  (previews (map-elt data 'url-previews)))
         (goto-char (point-max))
-        (cl-loop for i from 1 to num
-                 do (when-let ((inner-data (map-elt data (intern (concat "preview-" (number-to-string i))))))
-                      (unless (looking-at-p "^\n") (insert "\n"))
-                      (insert (ement-room--format-url-preview inner-data)))))
+        (dolist (preview previews)
```

Comment:
This is looking better.  Now it probably should go into its own function, similar to `ement-room--format-m.image`.  And the code that calls it should probably go in `ement-room--format-message-body`, treating it as a kind of "appendix" to the message body.

Reply:
I've currently left it how it is, mainly due to the fact that there is already a function named `ement-room--format-url-preview`. I think the most reasonable way to go about it is integrating the dolist into `ement-room--format-url-preview` and calling it from `ement-room--format-message-body`, but just wanted to confirm with you before I do it.

Reply:
I'll also probably implement the message parsing in this combined function to insert the previews in the correct order rather than just iterating using dolist.